### ISSUE TITLE
ControlledMenu not calling onClose when child gets updated

### ISSUE THREAD

**Issue by rortan134** [szhsin/react-menu#588]
React/React-dom version: ^18.2.0
React-Menu version: ^3.1.1

## Describe the bug

When using the controlled menu component with custom anchor points, the onClose prop responsible for closing the menu whenever needed does not get called after its child has just been updated (assumed after a series of tests).

## To Reproduce

1. Set up regular ControlledMenu with anchorPoint that gets set and open a contextMenu on right click
2. ControlledMenu child is a custom component with a bunch of state-decided menu items to display, for example, if you right-click on an element, render “delete element”, else render “add element”, and if you click outside change some state changing what gets rendered as a menu item and immediately closing.
3. Track the onClose callback

## Expected behavior

The onClose callback should always be called after clicking outside itself or doing a duplicate right-click.

## Code snippets/CodeSandbox examples/Screenshots

```
// Context Menu parent component

const [menuProps, toggleMenu] = useMenuState({ transition: true });
const [anchorPoint, setAnchorPoint] = useState(ORIGIN);

<ControlledMenu
                {...menuProps}
                anchorPoint={anchorPoint}
                position="initial"
                onClose={() => {
                    toggleMenu(false);
                    console.log("closed");
                }}
            >
                <ContextMenuOptions />
</ControlledMenu>
```

```
// Context menu options (child)

// zustand global state
const { selectedNode } = useTrackedNodes(); 
const { edgeCallback } = useTrackedUtils();

return (
        <>
            {edgeCallback ? (
                <MenuItem onClick={edgeCallback}>Delete connection</MenuItem>
            ) : isSelected() ? (
                <>
                    {singleNodeSelected && <ChangeTypeButton />}
                    <MenuDivider />
                    <MenuItem onClick={() => removeNode(selectedNode)}>
                      Delete Node
                    </MenuItem>
                </>
            ) : (
                <>
                    <MenuItem onClick={handleAddNode}>Add Node</MenuItem>
                    <MenuDivider />
                    <MenuItem onClick={clearNodes}>Clear</MenuItem>
                </>
            )}
        </>
    );

```

Whenever you click outside a node, the selected node is immediately set to undefined, making the isSelected() return false, changing the child's render and what i think probably causing the onClose to not get called.

```
onClose={() => {
                    toggleMenu(false);
                    console.log("closed");
                }}
```

It requires some attempts, but notice how sometimes it doesn't console log and the menu doesn't get closed and only gets the anchorPoint updated.
I am both right-clicking and left-clicking at the video.

https://user-images.githubusercontent.com/37253958/180606753-9c416c00-c7b5-4940-8d74-281cd85d1e8a.mp4

Please mind that I had to assume this bug based on tests, and I don't have a 100% way of reproducing it without some attempts.

**Comment by szhsin** [comment]
Hi @rortan134 , thanks for reporting the issue. 
Removing a focused menu item can cause menu to lose focus, therefore onClose event won't fire afterwards. I can confirm this is bug. Will fix it in the next release.

**Comment by szhsin** [comment]
Issue fixed by [v3.1.2](https://github.com/szhsin/react-menu/releases/tag/v3.1.2)
Example: https://codesandbox.io/s/react-menu-starter-forked-6m6g4x
